eclipse-theia/theia
# implement workbench: startup editor preference

#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/12754

This PR implements the workbench: startup editor preference. This allows the user to choose which editor is shown on startup when none are restored from the previous session.

Implemented preferences:
- `none`: no editor will be shown on startup.
- `welcomePage`: the default welcome page will be shown on startup.
- `readme`: it will open the folder's readme file in preview mode.
- `newUntitledFile`: opens an untitled file on startup.
- `welcomePageInEmptyWorkbench`: only opens the welcome page when opening
  empty workbench.


#### How to test


1.  Open Theia application.
2.  Go to `Settings` -> `workbench: startup editor` and select one of the preference to test.
3.  Close any opened editor and restart Theia.
4.  The selected editor should be shown on startup.
5.  Repeat 2-4 until all of the preferences are tested.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
